### ISSUE TITLE
Undefined index when editing a membergroup

### ISSUE THREAD

**Issue by live627** [elkarte/Elkarte#2789]
```
PHP Fatal error: Uncaught exception 'ErrorException' with message 'Undefined index: boardaccess' in /home/livecom/public_html/livemods.net/elk/sources/admin/ManageMembergroups.controller.php:603
Stack trace:
#0 /home/livecom/public_html/livemods.net/elk/sources/admin/ManageMembergroups.controller.php(603): error_handler(integer, string, string, integer, array)
#1 /home/livecom/public_html/livemods.net/elk/sources/subs/Action.class.php(175): action_edit()
#2 /home/livecom/public_html/livemods.net/elk/sources/admin/ManageMembergroups.controller.php(93): dispatch(string)
#3 /home/livecom/public_html/livemods.net/elk/sources/subs/Menu.subs.php(385): action_index()
#4 /home/livecom/public_html/livemods.net/elk/sources/admin/Admin.controller.php(78): callMenu(array)
#5 /home/livecom/public_html/livemods.net/elk/sources/SiteDispatcher.class.php(304): action_index()
#6 /home/livecom/public_html/livemods.net/elk/index.php(262): dispatch()
#7 /home/livecom/public_html/livemods.net/elk/index.php(192): elk_main()
#8 {main}
thrown in /home/livecom/public_html/livemods.net/elk/sources/admin/ManageMembergroups.controller.php on line 603
```
This is when editing a membergroup without any boards selected.

https://github.com/elkarte/Elkarte/blob/17216c0ec4d016448f5cb36b52f9308c04bc3f71/sources/admin/ManageMembergroups.controller.php#L603

**Comment by Vekseid** [comment]
Line should be
    if (!empty($our_post['boardaccess']))

**Comment by Spuds** [comment]
Mr 💩 the Christmas Poo!   

There are a bunch of problems there, caused (by me 👼)  when http req was used there.  Its been broken for a while too (like 18 months)

The code had been using __get methods from the validator which would return at least null.  However when I did the conversion I just used what was supplied in $_POST which is undefined for lots of things depending on what you are doing.  

I think all that is needed is an simple array_merge with null values for all things that were using __get, that should properly define the $our_post array.
